### PR TITLE
Allow setting near/far planes and listening for view update events

### LINKED ISSUE #32: Support updating near/far values
XRRenderState allows updating the near/far values, which get updated during the next rAF tick. We should probably provide an API for doing this (and calculate projection matrices accordingly)

### PR DESCRIPTION
**Pull request by Manishearth**
r? @asajeffrey

I couldn't handle magicleap, sadly, since I can't get it to build, and it seems to manage its own projection matrices

Fixes #32

### CODE REVIEW

File: webxr-api/view.rs [hunk 1]
```diff
@@ -90,3 +90,40 @@ pub enum Views {
...
     Stereo(View<LeftEye>, View<RightEye>),
 }
+
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "ipc", derive(Serialize, Deserialize))]
+pub struct ClipPlanes {
+    pub near: f32,
+    pub far: f32,
+    /// Was there an update that needs propagation to the client?
+    update: bool,
```
Comment (asajeffrey) [inline webxr-api/view.rs]: Hmm, it would be nice if we didn't have to keep a dirty bit on the `ClipPlanes` itself.

**Comment by jdm** [comment]
```
error[E0583]: file not found for module `utils`

  --> webxr/lib.rs:31:16

   |

31 | pub(crate) mod utils;

   |                ^^^^^

   |

   = help: name the file either utils.rs or utils/mod.rs inside the directory "webxr"

error: aborting due to previous error
```

**Pull request merged**